helix-term/src/ui/editor.rs

```diff
@@ -810,6 +814,29 @@ impl Component for EditorView {
                 status_msg,
                 style,
             );
+        } else {
+            let mut disp = String::new();
+            if let Some(count) = cx.editor.count {
+                disp.push_str(&count.get().to_string())
+            }
+            if let Some(keys) = self.keymaps.pending() {
+                for key in keys {
+                    let s = key.to_string();
+                    if s.width() > 1 {
+                        disp.push_str(&format!("<{}>", s));
+                    } else {
+                        disp.push_str(&s);
+                    }
+                }
+            }
+            let width = 10usize;
+            surface.set_string(
+                area.x + area.width.saturating_sub(width as u16),
+                area.y + area.height.saturating_sub(1),
+                disp.get(disp.len().saturating_sub(width)..)
+                    .unwrap_or(&disp),
```

Comment:
I don't think this is a good idea. We can get panic due to CJK characters being halfway through.

Reply:
It won't panic but instead the end will get chopped off instead of scrolling if the length is greater than the width. Doesn't `rposition()` take an element to search for ? We simply want the last 10 characters.

Reply:
I believe the `len()` here is len in terms of characters, which is not the same as text width occupied right? What will happen when a two width character is used? Maybe you can try `一` (one)? Can try create a key which can accept next key, and set the `get(width..)` to be an odd number, then it might break. If it doesn't then there should be no issue.

I am not sure about display but if the index is in unicode boundary it will return `None`, so we might display N characters, when N characters is more than the screen can handle then it might panic.

Reply:
> I am not sure about display but if the index is in unicode boundary it will return None, so we might display N characters, when N characters is more than the screen can handle then it might panic.

I checked and it doesn't panic, it simply chops off the remaining characters. So the problem here is that if any key is a multi byte character *and* the total width of the key display is greater than 15, there's a high chance that instead of scrolling to the right, only the characters from the beginning that fit within 15 cells will be shown; there is no panic.